GliaWidgets/ViewModel/Chat/ChatViewModel.swift

```diff
@@ -362,12 +362,24 @@ extension ChatViewModel {
             })?.offset
         else { return }
 
-        let status = Strings.Message.Status.delivered
-        let message = ChatMessage(with: message)
-        let item = ChatItem(kind: .visitorMessage(message, status: status))
-        downloader.addDownloads(for: message.attachment?.files, with: uploads)
+        let deliveredStatus = Strings.Message.Status.delivered
+
+        // Remove previous "Delivered" statuses
+        section.items
+            .enumerated()
+            .forEach { index, element in
+                if case .visitorMessage(let message, let status) = element.kind,
+                   status == deliveredStatus {
+                    let chatItem = ChatItem(kind: .visitorMessage(message, status: nil))
+                    section.replaceItem(at: index, with: chatItem)
+                }
+            }
+
+        let deliveredMessage = ChatMessage(with: message)
+        let item = ChatItem(kind: .visitorMessage(deliveredMessage, status: deliveredStatus))
+        downloader.addDownloads(for: deliveredMessage.attachment?.files, with: uploads)
         section.replaceItem(at: index, with: item)
-        action?(.refreshRow(index, in: section.index, animated: false))
+        action?(.refreshSection(section.index, animated: false))
```

Comment:
Have you considered just reloading the rows that need to be updated?

Reply:
That was the initial solution, yes. However, the animation behaved very weirdly for some reason, that's why I went with the entire section.

Reply:
The previous solution used `func refreshRow(_ row: Int, in section: Int, animated: Bool)` which in turn uses
`tableView.beginUpdates()  tableView.endUpdates()` but does not call `reloadRows` which seems odd

Reply:
I've changed it to only refresh the affected rows, actually calling the `reloadRows(at:with:)` between starting and ending the updates.